### ISSUE TITLE
Titles with special characters not matching

### ISSUE THREAD

**Issue by seanhoyt** [hulkrelax/hltb-for-deck#9]
It may be my imagination, but titles with special characters like ® (reg) and possibly other special characters won't match in HLTB. I'm seeing this with Titan Fall 2 and Tetris Effect: Connected.

**Comment by hulkrelax** [comment]
Yeah it's a bit of a mixed bag. I have to normalize the data on the steam side and sometimes that can cause issues. I try to remove special characters as part of the normalization process but there could be some issues. HLTB unfortunately doesn't have the best API (considering it's not obligated to) so I'm limited in my options. I'll try to look into it this weekend if I have some time.

**Comment by seanhoyt** [comment]
Does the HLTB API not support using the Steam App ID?

**Comment by hulkrelax** [comment]
HLTB isn't an open platform so they don't really have an endpoint designed for third parties. They aren't obligated to do that for anyone.

**Comment by woj-tek** [comment]
Was just playing with hltb-for-deck and noticed that LEGO games are not working... and it's the same issue due to `®` in the name :-)

**Comment by elkinjosetm** [comment]
I'm having this issue with Star Wars Jedi Fallen order, but on top of having an special character, I use my steam deck in Spanish, and for this particular game the "displayName" is slightly different due to the translation, so I wonder if this issue could actually be fixed by requesting the game info using the steamId with steam api, and then when you have the "original name" to use that text to try to match with the HLTB search endpoint you use.

**Comment by elkinjosetm** [comment]
Another alternative is to allow users try to manually match the game by implementing a search UI for HLTB, and letting users select from the result the game.

Similar to what SteamGridDB does, where they allow you to search/filter games and choose the image you want to use.

**Comment by hulkrelax** [comment]
I think I have fixed this for the most part. Special characters should get properly normalized now. I've tested this with games like Fallen Order as well as the Lego games (which has the reserve mark) and they stats show up now. Be on the lookout for 1.2.3 soonish

**Comment by anatolevuc** [comment]
Still no match for Rome II: Total War

**Comment by elkinjosetm** [comment]
Same here for **Star Wars: Jedi Fallen Order** in Spanish

**Comment by hulkrelax** [comment]
The "editions" are not surprising. Steam stores the game name as that so when searching by the name, HLTB returns no results. This bug fix addressed special characters, not where the game is "Special Edition" or "Ultimate Edition" or whatever. That is an issue I don't have a solution for at the moment because steam stores the game name as such and not just "Crysis 2" for example, it's stored as "Crysis 2 Maximum Edition" or whatever it is.

**Comment by hulkrelax** [comment]
> Same here for **Star Wars: Jedi Fallen Order** in Spanish

Could you elaborate here? There is a Spanish version of the game?

**Comment by elkinjosetm** [comment]
Yes, since I use my Steam Deck (Steam account) in Spanish, some games does have a different name, essentially the translation to whatever language.

Worth mentioning that not all games are the same, and in fact, personally I would like to have the original name instead, Steam doesn't let me choose/change the name.

That's why I suggested before if it was possible to implement a search to manually match the games, kinda like the SteamGridDB plugin does, where it let you search/filter by name and let you choose the asset you want to use, but for this plugin it would be more to fix the matching issue. That would even "fix" the issue with special characters you might find on some game names.

**Comment by anatolevuc** [comment]
> Yes, since I use my Steam Deck (and Steam account) in Spanish, some games does have a different name, essentially the translation to whatever language.
> 
> Worth mentioning that not all games are the same, and in fact, personally I would like to have the original name instead, Steam doesn't let me choose/change the name.
> 
> That's why I suggested before if it was possible to implement a search to manually match the games, kinda like the SteamGridDB plugin does, where it let you search/filter by name and let you choose the asset you want to use, but for this plugin it would be more to fix the matching issue. That would even "fix" the issue with special characters you might find on some game names.

Same for russian. For example Hogwarts Legacy called "Хогвартс. Наследие". Therefore no match

**Comment by alecchen** [comment]
> > Yes, since I use my Steam Deck (and Steam account) in Spanish, some games does have a different name, essentially the translation to whatever language.
> > Worth mentioning that not all games are the same, and in fact, personally I would like to have the original name instead, Steam doesn't let me choose/change the name.
> > That's why I suggested before if it was possible to implement a search to manually match the games, kinda like the SteamGridDB plugin does, where it let you search/filter by name and let you choose the asset you want to use, but for this plugin it would be more to fix the matching issue. That would even "fix" the issue with special characters you might find on some game names.
> 
> Same for russian. For example Hogwarts Legacy called "Хогвартс. Наследие". Therefore no match

I did some changes in #20 and it seems fixed the issue for non-english game title.

**Comment by Seraphli** [comment]
I have this problem with the game "Beyond: Two Souls"
